### PR TITLE
[ROCm] parse rocm version during hipify

### PR DESCRIPTION
**Pull request by jeffdaily**
### Description
The hipify mappings file now parses the rocm version header file and can use this information to conditionalize the mappings.  This is necessary while rocm packaging matures.

### Issue
#80849 updated the hipify mappings, but it wasn't backward compatible with ROCm versions prior to 5.2.

### Testing
Verified by building rocm pytorch using both the rocm 5.1 and 5.2 dockerfiles.

### CODE REVIEW

**Comment by acxz** [comment]
This PR also closes https://github.com/pytorch/pytorch/pulll/82252 created by @dmikushin

File: torch/utils/hipify/cuda_to_hip_mappings.py [hunk 1]
```diff
@@ -24,6 +27,34 @@
 supported in ROCm/HIP yet.
 """
 
+# We need to know the ROCm version so we can conditionalize some of the mappings later.
+# As of ROCm 5.0, the version is found in rocm_version.h header file under /opt/rocm/include.
+rocm_path = os.environ.get('ROCM_HOME') or os.environ.get('ROCM_PATH') or "/opt/rocm"
+try:
+    rocm_path = subprocess.check_output(["hipconfig", "--rocmpath"]).decode("utf-8")
+except subprocess.CalledProcessError:
+    print("Warining: hipconfig --rocmpath failed, assuming /opt/rocm")
```
Comment (jithunnair-amd) [inline torch/utils/hipify/cuda_to_hip_mappings.py:36]: nit:typo in "Warining"

**Review by albanD — APPROVED** [review]
Thanks for the quick fix.
The CI failures are relevant though!

**Pull request closed without merging**